app/models/filters/filter_base.rb

```diff
@@ -0,0 +1,383 @@
+    attribute :ph, Boolean, default: false
+
+    validates_presence_of :start, :end
+    validate do
+      if start > self.end
+        errors.add(:end, 'End date must follow start date.')
+      end
+    end
+
+    # NOTE: keep this up-to-date if adding additional attributes
+    def cache_key
+      [
+        user.id,
+        effective_project_ids,
+        cohort_ids,
+        coc_codes,
+        sub_population,
+        start_age,
+        end_age,
+      ]
+    end
+
+    # NOTE: params[:filters][:project_types] will be 'es', 'th', etc.
+    # the report expects @filter.project_types to be an array of integers 1, 2 etc.
+    def set_from_params(filters)
+      return unless filters.present?
+
+      self.start = filters.dig(:start)&.to_date
+      self.end = filters.dig(:end)&.to_date
+      pattern = filters.dig(:comparison_pattern)&.to_sym
+      self.comparison_pattern = pattern if PerformanceDashboards::Overview.valid_comparison_pattern?(pattern)
+      self.coc_codes = filters.dig(:coc_codes)&.select { |code| PerformanceDashboards::Overview.coc_codes.include?(code) }
+      self.household_type = filters.dig(:household_type)&.to_sym
+      self.hoh_only = filters.dig(:hoh_only) == '1'
+      self.project_type_codes = Array.wrap(filters.dig(:project_type_codes))&.reject { |type| type.blank? }.presence
+      self.data_source_ids = filters.dig(:data_source_ids)&.reject(&:blank?)&.map(&:to_i)
+      self.organization_ids = filters.dig(:organization_ids)&.reject(&:blank?)&.map(&:to_i)
+      self.project_ids = filters.dig(:project_ids)&.reject(&:blank?)&.map(&:to_i)
+      self.funder_ids = filters.dig(:funder_ids)&.reject(&:blank?)&.map(&:to_i)
+      self.veteran_statuses = filters.dig(:veteran_statuses)&.reject(&:blank?)&.map(&:to_i)
+      self.age_ranges = filters.dig(:age_ranges)&.reject(&:blank?)&.map { |range| range.to_sym }
+      self.genders = filters.dig(:genders)&.reject(&:blank?)&.map { |gender| gender.to_i }
+      self.sub_population = filters.dig(:sub_population)&.to_sym
+      self.races = filters.dig(:races)&.select { |race| HUD.races.keys.include?(race) }
+      self.ethnicities = filters.dig(:ethnicities)&.reject(&:blank?)&.map { |ethnicity| ethnicity.to_i }
+      self.project_group_ids = filters.dig(:project_group_ids)&.reject(&:blank?)&.map { |group| group.to_i }
+      self.prior_living_situation_ids = filters.dig(:prior_living_situation_ids)&.reject(&:blank?)&.map { |m| m.to_i }
+      self.destination_ids = filters.dig(:destination_ids)&.reject(&:blank?)&.map { |m| m.to_i }
+      self.length_of_times = filters.dig(:length_of_times)&.reject(&:blank?)&.map { |m| m.to_sym }
+      ensure_dates_work
+    end
+
+    def for_params
+      {
+        filters: {
+          start: start,
+          end: self.end,
+          comparison_pattern: comparison_pattern,
+          coc_codes: coc_codes,
+          household_type: household_type,
+          project_type_codes: project_type_codes,
```

Comment:
Not yet, but probably yes, eventually since not all types are currently represented with codes.  Alternatively, and maybe better, would be to setup a mapping for project type code to project type that allows for all types to be specified.